lib/active_merchant/billing/gateways/adyen.rb

```diff
@@ -558,7 +557,7 @@ def add_payment(post, payment, options, action = nil)
         elsif payment.is_a?(Check)
           add_bank_account(post, payment, options, action)
         else
-          add_mpi_data_for_network_tokenization_card(post, payment, options) if payment.is_a?(NetworkTokenizationCreditCard)
+          add_network_tokenization_card(post, payment, options) if payment.is_a?(NetworkTokenizationCreditCard) || options[:wallet_type] == :google_pay
```

Comment:
What is the situation where there will be an `options[:wallet_type]` but it won't be a `NetworkTokenizationCreditCard`?

Reply:
Ah, nevermind, I'm following this now.